src/Compilers/Core/Portable/Compilation/SymbolInfo.cs

```diff
@@ -69,18 +56,17 @@ internal SymbolInfo(ISymbol symbol, CandidateReason reason)
         }
 
         internal SymbolInfo(ImmutableArray<ISymbol> candidateSymbols, CandidateReason candidateReason)
-            : this(null, candidateSymbols, candidateReason)
+            : this(symbol: null, candidateSymbols, candidateReason)
         {
         }
 
-        internal SymbolInfo(ISymbol? symbol, ImmutableArray<ISymbol> candidateSymbols, CandidateReason candidateReason)
-            : this()
+        private SymbolInfo(ISymbol? symbol, ImmutableArray<ISymbol> candidateSymbols, CandidateReason candidateReason)
```

Comment:
this constructor was made private.  callers should call into one of hte prior constructors (which ensure appropriate invariants about this type).